code/datums/diseases/heart_failure.dm

```diff
@@ -0,0 +1,59 @@
+/datum/disease/heart_failure
+	form = "Condition"
+	name = "Myocardial Infarction"
+	max_stages = 5
+	stage_prob = 2
+	cure_text = "Heart replacement surgery to cure. Defibrillation (or as a last resort, uncontrolled electric shocking) may also be effective after the onset of cardiac arrest. Corazone can also mitigate cardiac arrest."
+	agent = "Shitty Heart"
+	viable_mobtypes = list(/mob/living/carbon)
+	permeability_mod = 1
+	desc = "If left untreated the subject will die!"
+	severity = "Dangerous!"
+	disease_flags = CAN_CARRY|CAN_RESIST
+	spread_flags = VIRUS_SPREAD_NON_CONTAGIOUS
+	visibility_flags = HIDDEN_PANDEMIC
+	required_organs = list(/obj/item/organ/heart)
+	var/sound = FALSE
+
+/datum/disease/heart_failure/stage_act()
+	..()
+	var/obj/item/organ/heart/O = affected_mob.getorgan(/obj/item/organ/heart)
+	var/mob/living/carbon/H = affected_mob
+	if(O && H.can_heartattack())
+		switch(stage)
+			if(1 to 2)
+				if(prob(2))
+					to_chat(H, "<span class='warning'>You feel [pick("discomfort", "pressure", "a burning sensation", "pain")] in your chest.</span>")
+				if(prob(2))
+					to_chat(H, "<span class='warning'>You feel dizzy.</span>")
+					H.confused += 6
+				if(prob(3))
+					to_chat(H, "<span class='warning'>You feel [pick("full", "nauseous", "sweaty", "weak", "tired", "short on breath", "uneasy")].</span>")
+			if(3 to 4)
+				if(!sound)
+					H.playsound_local(H, 'sound/health/slowbeat.ogg',40,0, channel = CHANNEL_HEARTBEAT)
+					sound = TRUE
+				if(prob(3))
+					to_chat(H, "<span class='danger'>You feel a sharp pain in your chest!</span>")
+					if(prob(25))
+						affected_mob.vomit(95)
+					H.emote("cough")
```

Comment:
this will make someone cough and vomit at the same time, which i'm not entirely sure is possible

Reply:
it certainly is in real life

Reply:
i'll take your word on it!

Reply:
as somebody with a long history of being sick I can confirm I have coughed and vomited at the same time